src/panels/config/zwave/zwave-network.ts

```diff
@@ -9,190 +10,233 @@ import {
+                                  "ui.panel.config.zwave.network_status.network_started_note_all_queried"
+                                )}
+                              </small>
+                            `
+                          : ""}
+                      `}
+                </div>
+                <div class="card-actions">
+                  ${this._networkStatus.state >= ZWAVE_NETWORK_STATE_AWAKED
+                    ? html`
+                        ${this._generateServiceButton("stop_network")}
+                        ${this._generateServiceButton("heal_network")}
+                        ${this._generateServiceButton("test_network")}
+                      `
+                    : html`
+                        ${this._generateServiceButton("start_network")}
+                      `}
+                </div>
+                ${this._networkStatus.state >= ZWAVE_NETWORK_STATE_AWAKED
+                  ? html`
+                      <div class="card-actions">
+                        ${this._generateServiceButton("soft_reset")}
+                        <ha-call-api-button
+                          .hass=${this.hass}
+                          path="zwave/saveconfig"
+                        >
+                          ${this.hass!.localize(
+                            "ui.panel.config.zwave.services.save_config"
+                          )}
+                        </ha-call-api-button>
+                      </div>
+                    `
+                  : ""}
+              </ha-card>
+              ${this._networkStatus.state >= ZWAVE_NETWORK_STATE_AWAKED
+                ? html`
+                    <ha-card class="content">
+                      <div class="card-actions">
+                        ${this._generateServiceButton("add_node_secure")}
+                        ${this._generateServiceButton("add_node")}
+                        ${this._generateServiceButton("remove_node")}
+                      </div>
+                      <div class="card-actions">
+                        ${this._generateServiceButton("cancel_command")}
+                      </div>
+                    </ha-card>
+                  `
+                : ""}
+            `
+          : ""}
       </ha-config-section>
     `;
   }
 
+  private async _getNetworkStatus(): Promise<void> {
+    this._networkStatus = await fetchNetworkStatus(this.hass!);
+  }
+
+  private _subscribe(): void {
+    this._unsubs = [
+      "zwave.network_start",
+      "zwave.network_stop",
+      "zwave.network_ready",
+      "zwave.network_complete",
+      "zwave.network_complete_some_dead",
+    ].map((e) =>
+      this.hass!.connection.subscribeEvents(
+        (event) => this._handleEvent(event),
+        e
+      )
+    );
+  }
+
+  private _unsubscribe(): void {
+    while (this._unsubs.length) {
+      this._unsubs.pop()!.then((unsub) => unsub());
+    }
+  }
+
+  private _handleEvent(event) {
+    this._getNetworkStatus();
```

Comment:
Why do we need to fetch the network status, can't we derive it from the event type like you're doing on the line below ?

Reply:
In its current form that would work, but I'm planning to include additional information in the data returned from the _getNetworkStatus() call that we can't derive from those events.

Reply:
the problem now is that there is not a single source of truth for the network status. Instead of keeping track of a boolean, I would suggest you extract the actual status from the status object, and have events optimistically set this. That way there is only 1 variable that we should track.